code/modules/mob/living/simple_animal/friendly/mule_bot/mule_bot_modules.dm

```diff
@@ -0,0 +1,182 @@
+//towing moduke -- uncertain
+//pepperball module -- might make
+//motion detector module -- will make
+//auto flare mod -- finished
+//medical mod -- uncertain
+//personal storage module -- finished
+//storage module -- finished
+
+/obj/item/mule_module
+	name = "mule module"
+	desc = "Can be used to improve a mule and specialize it"
+	icon = 'icons/obj/items/items.dmi'
+	icon_state = "modkit"
+	var/mob/living/simple_animal/mule_bot/attached_mule
+	var/mutable_appearance/mod_overlay
+	var/overlay_icon = 'icons/mob/kerfus.dmi'
+	var/overlay_icon_state = "backpack"
+	var/y_offset = 0
+	var/x_offset = 0
+	var/module_desc = "this module does not do that much, you think. You'd have no way of knowing."
+
+
+/obj/item/mule_module/Initialize()
+	. = ..()
+	mod_overlay = mutable_appearance(overlay_icon, overlay_icon_state, BACK_LAYER, FLOAT_PLANE)
+	mod_overlay.pixel_y = y_offset
+	mod_overlay.pixel_x = x_offset
+
+/obj/item/mule_module/proc/apply(mob/living/simple_animal/mule_bot/mule)
+	RegisterSignal(mule, COMSIG_PARENT_EXAMINE, PROC_REF(examine_parent))
+	mule.installed_module = src
+	attached_mule = mule
+	src.forceMove(mule)
```

Comment:
This applies to the entire PR: "src.myvar" is redundant just use "myvar" directly